FirebaseStorage/Sources/StorageUploadTask.swift

```diff
@@ -36,21 +36,20 @@ import Foundation
    * Prepares a task and begins execution.
    */
   @objc open func enqueue() {
-    weak var weakSelf = self
-    dispatchQueue.async {
-      guard let strongSelf = weakSelf else { return }
-      if let contentValidationError = strongSelf.contentUploadError() {
-        strongSelf.error = contentValidationError
-        strongSelf.finishTaskWithStatus(status: .failure, snapshot: strongSelf.snapshot)
+    // Capturing self so that the upload is done whether or not there is a callback.
+    dispatchQueue.async { [self] in
```

Comment:
optional nit: `[self] in` is the default capture and can be omitted. I'm fine with leaving this as-is though.

Reply:
Understood. I wanted to be explicit to contrast with the other files.